### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by CandyRedTele**

### CODE REVIEW

File: src/space-commander/space-commander-main.cpp [hunk 1]
```diff
@@ -237,3 +240,53 @@ void out_of_memory_handler()
...
+            Shakespeare::log(Shakespeare::ERROR, LOGNAME, log_buffer);
+            isValid = false;
+        }
+    }
+
+    if (!isValid) {
+        memset(log_buffer,0, MAX_BUFFER_SIZE);
+        snprintf(log_buffer, MAX_BUFFER_SIZE, "exiting");
+        Shakespeare::log(Shakespeare::ERROR, LOGNAME, log_buffer);
+        exit(1);
```
Comment (SpaceShawn) [inline src/space-commander/space-commander-main.cpp]: Let's avoid exiting the commander where possible. We may never be able to establish communication again if the commander keeps exiting and being restarted by PCD. We could return these invalid states (i.e. write some generic error buffer to the pipes in order to reach the ground) so we can know the problem without restarting the commander

File: src/space-commander/space-commander-main.cpp [hunk 2]
```diff
@@ -25,6 +27,7 @@ const char ERROR_EXECUTING_COMMAND = '2';
 // Declarations
 static void out_of_memory_handler();
 static int perform(int bytes);
+static void validate();
```
Comment (SpaceShawn) [inline src/space-commander/space-commander-main.cpp]: Let's have validate return an integer representing errorcodes [(link)](https://github.com/spaceconcordia/space-lib/blob/master/include/SpaceDecl.h#L118-126) and have the commands calling validate return them to the ground

Comment (CandyRedTele) [inline src/space-commander/space-commander-main.cpp]: I agree that the commands should do validation and return status, but this specific validate function is meant to run at start-up and log warnings if the state of the commander is not as expected.
I removed the exit(1) to simply log an unstable state error... so at least we'll have some clues of what is going on

Comment (SpaceShawn) [inline src/space-commander/space-commander-main.cpp]: Okay sounds good


*Shawn Bulger*

*ConSat-1, **Ground Station*


 Space Concordia <http://www.spaceconcordia.ca>


*H1029-7, *1455 Rue De Maisonneuve West

*Montréal, Québec **H3G 1M8*

(514) 848-2424 ext. 2738

File: src/space-commander/space-commander-main.cpp (hunk 1 above)
Comment (CandyRedTele) [inline src/space-commander/space-commander-main.cpp]: Totally true ... changing this.

**Pull request merged**